packages/itwin/grouping-mapping-widget/src/widget/components/provider/CustomUIProvider.ts

```diff
@@ -0,0 +1,45 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { ReactElement } from "react";
+
+/**
+ * Custom UI provider type definition.
+ */
+export interface CustomUIProvider {
```

Comment:
do you think tjis will confuse users who are already familiar with ui provider from core? is there a better name for this?

Reply:
That's why it's called custom UI provider and it is only used when configuring grouping mapping widget.

Reply:
i dont think adding custom is enough, thats too generic

Reply:
hmm any suggestions?

Reply:
sorry no

Reply:
How about GMCustomUIProvider?

Reply:
better, thats on your team to decide tho

Reply:
We have discussed about this and chose this name after the 'extension' name was rejected.

Reply:
what was this renamed to??

Reply:
GroupingMappingUIProvider, will push change soon

Reply:
not to be confused with `GroupingMappingProvider` which implements `UiItemsProvider` ?

Reply:
Done

Reply:
I really don't think this rename is enough. I can see this being very confusing if core defines what a UI provider is and you guys piggy back off that name and change it entirely. 

@arnobmallickbsw @Guillar1 @MindaugasButkus can you guys help come up with a less confusing name?

Reply:
GroupingMappingCustomUI? Or GroupingMappingWidgetUI
I prefer the first one.

Reply:
the first one sounds much better than what we have now, so lets go with that. pleasse replace any usage of Uiprovider thats not pertinent to the UiItemsProvider. so make sure you scrub the changelog file as well.